client/FxAccountClient.js

```diff
@@ -619,7 +619,40 @@ define([
   FxAccountClient.prototype.accountStatus = function(uid) {
     required(uid, 'uid');
 
-    return this.request.send('/account/status?uid=' + uid, 'GET');
+    return this.accountStatusFromOptions({uid: uid});
+  };
+
+  /**
+   * Gets the status of an account from options.
+   *
+   * @method accountStatusFromOptions
+   * @param {Object} [options={}] Options
+   *   @param {String} [options.uid]
+   *   Use the uid to get account status
+   *   @param {String} [options.email]
+   *   Use email to get account status
+   * @return {Promise} A promise that will be fulfilled with JSON `xhr.responseText` of the request
+   */
+  FxAccountClient.prototype.accountStatusFromOptions = function(options) {
```

Comment:
I know checking the status of both a uid and email call the same endpoint, the current approach feels like it jams two separate concepts into a single function. Maybe revert the original `accountStatus` and add a new function that's `emailStatus` or similar?